store/store_test.go

```diff
@@ -2486,3 +2486,174 @@ func (t *remoteRepoTestSuite) TestUbuntuStorePaymentMethodsHandles401(c *C) {
 	c.Assert(err, NotNil)
 	c.Check(err.Error(), Equals, "invalid credentials")
 }
+
+const customerJsonTosAccepted = `
+{
+  "latest_tos_date": "2016-09-14T00:00:00+00:00",
+  "accepted_tos_date": "2016-09-14T15:56:49+00:00",
+  "latest_tos_accepted": true,
+  "has_payment_method": true
+}
+`
+
+func (t *remoteRepoTestSuite) TestUbuntuStoreReadyToBuy(c *C) {
+	purchaseServerGetCalled := 0
+	mockPurchasesServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case "GET":
+			// check device authorization is set, implicitly checking doRequest was used
+			c.Check(r.Header.Get("X-Device-Authorization"), Equals, `Macaroon root="device-macaroon"`)
+
+			c.Check(r.Header.Get("Authorization"), Equals, t.expectedAuthorization(c, t.user))
+			c.Check(r.URL.Path, Equals, "/purchases/v1/customers/me")
+			io.WriteString(w, customerJsonTosAccepted)
+			purchaseServerGetCalled++
+		default:
+			c.Error("Unexpected request method: ", r.Method)
+		}
+	}))
+
+	c.Assert(mockPurchasesServer, NotNil)
+	defer mockPurchasesServer.Close()
+
+	customersMeURI, err := url.Parse(mockPurchasesServer.URL + "/purchases/v1/customers/me")
+	c.Assert(err, IsNil)
+
+	authContext := &testAuthContext{c: c, device: t.device, user: t.user}
+	cfg := Config{
+		CustomersMeURI: customersMeURI,
+	}
+	repo := New(&cfg, authContext)
+	c.Assert(repo, NotNil)
+
+	err = repo.ReadyToBuy(t.user)
+	c.Check(err, IsNil)
+	c.Check(purchaseServerGetCalled, Equals, 1)
+}
+
+const customerJsonTosNotAccepted = `
+{
+  "latest_tos_date": "2016-10-14T00:00:00+00:00",
+  "accepted_tos_date": "2016-09-14T15:56:49+00:00",
+  "latest_tos_accepted": false,
+  "has_payment_method": true
+}
+`
+
+func (t *remoteRepoTestSuite) TestUbuntuStoreReadyToBuyNotAcceptedTos(c *C) {
+	purchaseServerGetCalled := 0
+	mockPurchasesServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
On a quick scan, those tests all look exactly the same. Copy & pasting all the logic with small variations increases the maintenance burden, and makes it uncomfortable to find the real meat of each test.

I'd recommend extracting what's not common between them into a table such as:
```
var readyToBuyTests = []readyToBuyTest{{
    foo: 1,
    bar, 2,
}, {
    foo: 3,
    bar: 4,
}}
```
and then iterating over that slice in a single test to perform the checks over the different scenarios.